hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java

```diff
@@ -150,6 +142,49 @@ public static void verifyResourceName(String resName)
       throw new IllegalArgumentException("Bucket or Volume name "
           + "cannot end with a period or dash");
     }
+  }
+
+  private static boolean isLowercaseAlphanumeric(char c) {
+    return (Character.toString(c).matches("[a-z0-9]"));
```

Comment:
I think this can be improved:

1. `doNameChecks` can verify that `resName` `matches()` the pattern `[a-z0-9.-]+`, instead of checking each character separately in `doCharacterChecks`
2. `Pattern` can be pre-compiled as a constant

Reply:
We can avoid the string conversion and pattern match with the following:
return Character.isLowerCase(c) || Character.isDigit(c)

Reply:
Thanks for the review and suggestion @adoroszlai . I have removed the pattern match code altogether. Made the change as suggested in the comment from @xiaoyuyao

Reply:
Sure, that's fine, too.

Reply:
Unless Java inlines the isLower or isDigit, from the performance point of view, I bet that checking if (char < a || char > 9) is way more faster. If someone can profile and it prove that I am wrong, I would be very happy to learn how Java works. But from a gut perspective I think this is actually slower -- but then again, it really does not matter, and if it improves code reading I am all for it. Just commenting here for posterity's sake; so that someone profiles this and tells us the perf argument did not hold.